### DISCUSSION TITLE
Docker MPD question about Music and Playlist directory  (Music directory not found)

### DISCUSSION THREAD

**Discussion by cequalspid** [jcorporation/myMPD#1315] [Q&A]
Question about the music and playlist directory. myMPD is working and I can see and use my playlists and all my artists/albums list under Browse on myMPD. But myMPD keeps showing two things:

_Music directory not found
Updating caches_

When setting option under <...>MPD connection --> music directory,  neither 'Autodetect", nor 'Specifiy' are accepted.  Should I just set to 'None'? Or change Docker below settings?

My _mpd .conf_ has the following directories:
     music_directory	 "/mounts/musicplus/flac_music"
     playlist_directory	  "~/.config/mpd/playlists"

My compose file (from https://jcorporation.github.io/myMPD/installation/docker):
      - /var/lib/mpd/music:/var/lib/mpd/music:ro  
      - /var/lib/mpd/playlists:/var/lib/mpd/playlists:ro  

/var/lib/mpd/music and /var/lib/mpd/music exist (owned by mpd:mpd) and are empty.

![20h_43m_57s_21_June_2024](https://github.com/jcorporation/myMPD/assets/102390165/f53a1881-68d4-4374-8da9-44638812b39d)

**Comment by jcorporation** [comment]
### 1) Music directory not found

This is mostly a problem with permissions. The myMPD log should show details.

- You use an outdated compose file 
- PUID and PGID are not used by the myMPD container image.
- If you want to use "autodetect" the paths inside and outside the container for music dir and playlist dir must be the same
- myMPD requires the r+x permissions for this directories.

### 2) Updating caches

Without logs I can not tell you what is going on. Attaching verbose logs is always a good idea ;)

**Comment by cequalspid** [comment]
I don't think it has to do with the compose file, six other containers are running. But I  removed PUID/GUID and replaced with 'user 1000:1000'.

I was thinking about what you said re:permisssions and realized I was starting mpd with systemd which runs it as mpd:mpd. So I removed that service and instead autostart with local bash_profile. 

Now mpd runs as 1000:1000. So that's out of the way. I also chown'ed var/lib/mpd to user 1000:1000.
All local mpd directories are owned by 1000:1000.  I can play music with ncmpcpp/mpd with no issues.

Now when I try to connect via the web interface 127.0.0.1:8082 (chosen because 8080 was already being used) it only shows 'Initializing myMPD'.  It seems myMPD won't connect to the localhost mpd and I cannot figure out why. (I have no password set in mpd.conf)

Docker logs:

> NOTICE   mympd     Cache dir: "/var/cache/mympd"
> NOTICE   mympd     Work dir: "/var/lib/mympd"
> NOTICE   mympd     Config dir: "/var/lib/mympd/config"
> NOTICE   mympd     Setting loglevel to NOTICE
> NOTICE   mympd     Starting myMPD 15.0.2
> NOTICE   mympd     Listening on http://[::]:8082
> NOTICE   mympd     Serving files from "/var/lib/mympd/empty"
> NOTICE   mympd     Empty dir: "/var/lib/mympd/empty"
> NOTICE   mympd     Pics dir: "/var/lib/mympd/pics"
> NOTICE   mympd     Backgrounds dir: "/var/lib/mympd/pics/backgrounds"
> NOTICE   mympd     Thumbnails dir: "/var/lib/mympd/pics/thumbs"
> NOTICE   mympd     Scripts dir: "/var/lib/mympd/scripts"
> NOTICE   mympd     Smartpls dir: "/var/lib/mympd/smartpls"
> NOTICE   mympd     State dir: "/var/lib/mympd/state"
> NOTICE   mympd     Default partition dir: "/var/lib/mympd/state/default"
> NOTICE   mympd     Tags cache dir: "/var/lib/mympd/tags"
> NOTICE   mympd     Webradio dir: "/var/lib/mympd/webradios"
> NOTICE   mympd     Covercache dir: "/var/cache/mympd/covercache"
> NOTICE   mympd     Webradiodb cache dir: "/var/cache/mympd/webradiodb"
> NOTICE   mympd     Starting mympd api thread
> NOTICE   mympd     Starting webserver thread
> NOTICE   mympd     myMPD is ready
> NOTICE   mympdapi  Reading global states
> NOTICE   mympdapi  "default": Reading partition states from directory "/var/lib/mympd/state/default"
> NOTICE   mympdapi  "stickerdb": Connecting to "localhost:6600"
> ERROR    mympdapi  "stickerdb": Connection: Connection refused
> NOTICE   mympdapi  "default": Connecting to "localhost:6600"
> ERROR    mympdapi  "default": Connection: Connection refused
> NOTICE   mympdapi  "default": Connecting to "localhost:6600"
> ERROR    mympdapi  "default": Connection: Connection refused
> NOTICE   mympdapi  "default": Connecting to "localhost:6600"
> ERROR    mympdapi  "default": Connection: Connection refused
> NOTICE   mympdapi  "default": Connecting to "localhost:6600"
> ERROR    mympdapi  "default": Connection: Connection refused
> NOTICE   mympdapi  "default": Connecting to "localhost:6600"
> ERROR    mympdapi  "default": Connection: Connection refused
> NOTICE   mympdapi  "default": Connecting to "localhost:6600"
> ERROR    mympdapi  "default": Connection: Connection refused
> NOTICE   mympdapi  "default": Connecting to "localhost:6600"
> ERROR    mympdapi  "default": Connection: Connection refused
> NOTICE   mympdapi  "default": Connecting to "localhost:6600"
> ERROR    mympdapi  "default": Connection: Connection refused
> NOTICE   mympdapi  "default": Connecting to "localhost:6600"
> ERROR    mympdapi  "default": Connection: Connection refused
> NOTICE   mympdapi  "default": Connecting to "localhost:6600"
> ERROR    mympdapi  "default": Connection: Connection refused
> NOTICE   mympdapi  "default": Connecting to "localhost:6600"
> ERROR    mympdapi  "default": Connection: Connection refused
> NOTICE   mympdapi  "default": Connecting to "localhost:6600"
> ERROR    mympdapi  "default": Connection: Connection refused
> NOTICE   mympdapi  "default": Connecting to "localhost:6600"
> ERROR    mympdapi  "default": Connection: Connection refused
> NOTICE   mympdapi  Starting mpd_worker thread for MYMPD_API_COVERCACHE_CROP
> NOTICE   mpdworker Deleted 0 files from covercache
> NOTICE   mpdworker Stopping mpd_worker thread
> NOTICE   mympdapi  "default": Connecting to "localhost:6600"
> ERROR    mympdapi  "default": Connection: Connection refused
> NOTICE   mympdapi  "default": Connecting to "localhost:6600"
> ERROR    mympdapi  "default": Connection: Connection refused
> NOTICE   mympdapi  "default": Connecting to "localhost:6600"
> ERROR    mympdapi  "default": Connection: Connection refused
> NOTICE   mympdapi  "default": Connecting to "localhost:6600"
> ERROR    mympdapi  "default": Connection: Connection refused

**Comment by cequalspid** [comment]
Not that this will help anyone else, but I got it working using local myMPD install. 
I think the connection refused likely had to do with permission issues, or ALSA access issues but I gave up on the docker install.

**Reply by jcorporation** [reply]
If mpd and myMPD runs on the same host a socket connection is preferable. You can simply mount the mpd socket into the myMPD container. I am not a docker expert, but it seems connecting from a docker container to localhost does not work.

The myMPD is initializing message is strange, this was already fixed a few versions ago. I will test this in my setup.
